samcli/lib/sync/infra_sync_executor.py

```diff
@@ -55,6 +55,27 @@
 }
 
 
+class InfraSyncResult:
+    """Data class for storing infra sync result"""
+
+    _infra_sync_executed: bool
+    _code_sync_resources: Set[ResourceIdentifier]
+
+    def __init__(self, executed: bool, code_sync_resources: Set[ResourceIdentifier] = set()) -> None:
+        self._infra_sync_executed = executed
+        self._code_sync_resources = code_sync_resources
+
+    @property
+    def infra_sync_executed(self) -> bool:
+        """Returns a boolean indicating whether infra sync executed"""
+        return self._infra_sync_executed
+
+    @property
+    def code_sync_resources(self) -> Set[ResourceIdentifier]:
+        """Returns a set of resource identifiers that need an infra sync"""
+        return self._code_sync_resources
```

Comment:
We could annotate this class with `@dataclass` and remove all the methods which is been added here

Reply:
But we need to encapsulate the fields to only be readable after instantiation, so I don't think we can remove any methods here